### PR TITLE
Update Fortran combinational logic openmp-primes sample

### PR DESCRIPTION
**Pull request by krisdale**
# Existing Sample Changes
## Description

- Update sample to work with ifx
- Formatting updates
- Minor editorial updates

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: DirectProgramming/Fortran/CombinationalLogic/openmp-primes/sample.json [hunk 1]
```diff
@@ -3,23 +3,23 @@
   "name": "OpenMP* Primes",
   "categories": ["Toolkit/oneAPI Direct Programming/Fortran/Combinational Logic"],
   "description": "Fortran Tutorial - Using OpenMP* (OMP)",
-  "toolchain": [ "ifort" ],
+  "toolchain": [ "ifx" ],
   "languages": [ { "fortran": {} } ],
   "targetDevice": [ "CPU" ],
   "os": [ "darwin" ],
```
Comment (jimmytwei) [inline DirectProgramming/Fortran/CombinationalLogic/openmp-primes/sample.json:9]: @krisdale Can you remove ``darwin`` from os and add in both ``linux`` and ``windows``?
Ex: ``"os": ["linux", "windows"],``

File: DirectProgramming/Fortran/CombinationalLogic/openmp-primes/sample.json [hunk 2]
```diff
@@ -3,23 +3,23 @@
...
   "builder": [ "make" ],
   "ciTests":{
     "darwin": [
```
Comment (jimmytwei) [inline DirectProgramming/Fortran/CombinationalLogic/openmp-primes/sample.json:12]: @krisdale Can you add the build steps for Linux and Windows? Remove ``darwin`` and add ``linux`` and ``windows``.

**Comment by krisdale** [comment]
Thanks for the review @jimmytwei  - updates applied.

**Review by jimmytwei — APPROVED** [review]

**Comment by jimmytwei** [comment]
@krisdale Approved. Sample works with the latest (7/31/23) 2024.0 BaseKit and HPCKit. Merging into development branch.

**Pull request merged**